src/components/axes/timeAxis.ts

```diff
@@ -280,12 +287,19 @@ export module Axis {
 
     protected _setup() {
       super._setup();
+      this._createDomElements();
+    }
+
+    private _createDomElements() {
+      d3.selectAll(".time-axis-tier").remove();
+
       this._tierLabelContainers = [];
       this._tierMarkContainers = [];
       this._tierBaselines = [];
       this._tickLabelContainer.remove();
       this._baseline.remove();
-      for (var i = 0; i < Time._NUM_TIERS; ++i) {
+
+      for (var i = 0; i < this._numTiers; ++i) {
         var tierContainer = this._content.append("g").classed("time-axis-tier", true);
         this._tierLabelContainers.push(tierContainer.append("g").classed(AbstractAxis.TICK_LABEL_CLASS + "-container", true));
         this._tierMarkContainers.push(tierContainer.append("g").classed(AbstractAxis.TICK_MARK_CLASS + "-container", true));
```

Comment:
As an aside, why not use D3 to append the "g"'s rather than us manually doing it?